evm/db/state.py

```diff
@@ -153,6 +153,9 @@ def delete_account(self, address):
     def account_exists(self, address):
         return bool(self._trie[address])
 
+    def account_has_code_or_nonce(self, address):
+        return self.get_code_hash(address) != EMPTY_SHA3 or self.get_nonce(address) != 0
```

Comment:
trivial but inverting these checks has a performance gain that we should probably take advantage of since this can end up being called numerous times during txn execution.


- `self.get_code_hash`  - 2 tries lookups and a keccak
- `self.get_nonce` - 1 trie lookup